pipeline/main.py

```diff
@@ -1,279 +0,0 @@
-    # Step 1: Prepare graph data and retrieve train/validation/test index ============================= #
-    # Load from DGL dataset
-    if args.dataset == 'Cora':
-        dataset = CoraGraphDataset(raw_dir=args.raw_dir)
-    elif args.dataset == 'Citeseer':
-        dataset = CiteseerGraphDataset(raw_dir=args.raw_dir)
-    elif args.dataset == 'Pubmed':
-        dataset = PubmedGraphDataset(raw_dir=args.raw_dir)
-    else:
-        raise ValueError('Dataset {} is invalid.'.format(args.raw_dir))
-
-    graph = dataset[0]
-    graph = dgl.add_self_loop(graph)
-
-    # check cuda
-    if args.gpu > 0 and torch.cuda.is_available():
-        device = 'cuda:{}'.format(args.gpu)
-    else:
-        device = 'cpu'
-
-    # retrieve the number of classes
-    n_classes = dataset.num_classes
-
-    # retrieve labels of ground truth
-    labels = graph.ndata.pop('label').to(device).long()
-
-    # Extract node features
-    feats = graph.ndata.pop('feat').to(device)
-    n_features = feats.shape[-1]
-
-    # retrieve masks for train/validation/test
-    train_mask = graph.ndata.pop('train_mask')
-    val_mask = graph.ndata.pop('val_mask')
-    test_mask = graph.ndata.pop('test_mask')
-
-    train_idx = torch.nonzero(train_mask, as_tuple=False).squeeze().to(device)
-    val_idx = torch.nonzero(val_mask, as_tuple=False).squeeze().to(device)
-    test_idx = torch.nonzero(test_mask, as_tuple=False).squeeze().to(device)
-
-    graph.to(device)
-
-    # Step 2: Create model =================================================================== #
-    mixhopgcn_model = MixHopGCN(in_dim=n_features,
-                                hid_dim=args.hid_dim,
-                                out_dim=n_classes,
-                                num_layers=args.num_layers,
-                                p=args.p,
-                                input_dropout=args.input_dropout,
-                                layer_dropout=args.layer_dropout,
-                                activation=torch.tanh,
-                                batchnorm=True)
-    
-    mixhopgcn_model = mixhopgcn_model.to(device)
-
-    # Step 3: Create training components ===================================================== #
-    loss_fn = nn.CrossEntropyLoss()
-    opt = optim.SGD(mixhopgcn_model.parameters(), lr=args.lr, weight_decay=args.lamb)
-    scheduler = optim.lr_scheduler.StepLR(opt, args.step_size, gamma=args.gamma)
-
-    # Step 4: training epoches =============================================================== #
-    acc = 0
-    no_improvement = 0
-    epochs = trange(args.epochs, desc='Accuracy & Loss')
-    
-    for _ in epochs:
-
-        # Training and validation using a full graph
-        mixhopgcn_model.train()
-
-        logits = mixhopgcn_model.forward(graph, feats)
-
-        # compute loss
-        train_loss = loss_fn(logits[train_idx], labels[train_idx])
-        train_acc = torch.sum(logits[train_idx].argmax(dim=1) == labels[train_idx]).item() / len(train_idx)
-
```

Comment:
You want to disable gradient for the validation set using `with torch.no_grad():`.

Reply:
Done.